src/components/Router.js

```diff
@@ -45,24 +52,41 @@ class Router extends Component {
       .catch(() => log('Not signed in'))
   }
 
-  componentDidMount() {
-   getEvents()
+  // User needs to be checked before the page physically renders
+  // (otherwise, the login page will initially show on every refresh)
+  componentWillMount() {
```

Comment:
Does this need to be `willMount`? I think the `loaded` state will block rendering a login page regardless, and `willMount` is technically [deprecated](https://reactjs.org/docs/react-component.html#legacy-lifecycle-methods)

Reply:
True true